### PR TITLE
feat: Add custom sort to tickets list (#792)

### LINKED ISSUE #792: Implement sorting in tickets list
**Is it a ?**

- [ ] Bug Report

- [x] Feature Request
    
- [ ] Chore
    
### Bug Report    

**Actual Behaviour**

No custom sorting feature is avaliable in tickets activity

**Expected Behaviour**

Custom sorting feature should be there in tickets list

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by AkshayCHD**
Fixes #792 

Changes: Add feature to sort tickets list by name

Screenshots for the change: 
screenshot_1521778779

### CODE REVIEW

**Comment by AkshayCHD** [comment]
@iamareebjamal please review

File: app/src/main/java/org/fossasia/openevent/app/module/ticket/list/TicketsAdapter.java [hunk 1]
```diff
@@ -49,12 +52,24 @@ public HeaderViewHolder onCreateHeaderViewHolder(ViewGroup viewGroup) {
 
     @Override
     public void onBindHeaderViewHolder(HeaderViewHolder headerViewHolder, int position) {
-        headerViewHolder.bindHeader(tickets.get(position).getType());
+        if (sortByName) {
+            headerViewHolder.bindHeader(tickets.get(position).getName().substring(0, 1).toUpperCase(Locale.getDefault()));
+        } else {
+            headerViewHolder.bindHeader(tickets.get(position).getType());
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/module/ticket/list/TicketsAdapter.java]: Create a common function which returns a string called `getHeader`

File: app/src/main/java/org/fossasia/openevent/app/module/ticket/list/TicketsAdapter.java [hunk 2]
```diff
@@ -49,12 +52,24 @@ public HeaderViewHolder onCreateHeaderViewHolder(ViewGroup viewGroup) {
...
     }
 
     @Override
     public long getHeaderId(int position) {
-        return tickets.get(position).getType().hashCode();
+        if (sortByName) {
+            return tickets.get(position).getName().substring(0, 1).toUpperCase(Locale.getDefault()).hashCode();
+        } else {
+            return tickets.get(position).getType().hashCode();
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/module/ticket/list/TicketsAdapter.java]: Use `getHeader().hashCode()`

**Comment by iamareebjamal** [comment]
Also, first complete your previously assigned issues. These issues are very low priority. I see no benefit of adding sort by name to tickets

**Comment by AkshayCHD** [comment]
@iamareebjamal I think then it is better to sort tickets list on the basis of sold tickets

**Pull request closed without merging**